components/eam/src/chemistry/aerosol/wetdep.F90

```diff
@@ -502,798 +471,732 @@ subroutine wetdepa_v2( ncol, deltat, &
+                        resusp_x                                ) ! out
+
+! ------------------------------------------------------------------------------
+! do precip production, resuspension and scavenging
+! ------------------------------------------------------------------------------
+   integer, intent(in) :: is_st_cu ! options for stratiform (1) or convective (2) clouds
+                                        ! raindrop size distribution is
+                                        ! different for different cloud:
+                                        ! 1: assume marshall-palmer distribution
+                                        ! 2: assume log-normal distribution
+   integer, intent(in) :: mam_prevap_resusp_optcc       ! suspension options
+   real(r8),intent(in) :: pdel_ik       ! pressure thikness at current column and level [Pa]
+   real(r8),intent(in) :: evapx         ! evaporation at current layer [kg/kg/s]
+   real(r8),intent(in) :: precabx_base_old ! input of precipitation at cloud base [kg/m2/s]
+   real(r8),intent(in) :: precabx_old ! input of precipitation above this layer [kg/m2/s]
+   real(r8),intent(in) :: scavabx_old ! input of scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(in) :: precnumx_base_old ! input of precipitation number at cloud base [#/m2/s]
+   real(r8),intent(out) :: precabx_base_new ! output of precipitation at cloud base [kg/m2/s]
+   real(r8),intent(out) :: precabx_new ! output of precipitation above this layer [kg/m2/s]
+   real(r8),intent(out) :: scavabx_new ! output of scavenged tracer flux from above [kg/m2/s]
+   real(r8),intent(out) :: precnumx_base_new ! output of precipitation number at cloud base [#/m2/s]
+   real(r8),intent(out) :: resusp_x    ! aerosol mass re-suspension in a particular layer [kg/m2/s]
+
+
+   ! local variables
+   real(r8)            :: tmpa ! temporary working variable
+   real(r8), parameter :: prec_smallaa = 1.0e-30_r8  ! 1e-30 kg/m2/s (or mm/s) = 3.2e-23 mm/yr
+
+   ! initiate *_new in case they are not calculated
+   scavabx_new = scavabx_old
+   precnumx_base_new = precnumx_base_old
+   precabx_base_new = precabx_base_old
+
+   tmpa = max( 0.0_r8, evapx*pdel_ik/gravit )
+   precabx_new = min_max_bound(0.0_r8,precabx_base_new, precabx_old-tmpa)
+
+   if (precabx_new < prec_smallaa) then
+      ! precip rate is essentially zero so do complete resuspension
+      call wetdep_resusp_noprecip(                         &
+                   is_st_cu,       mam_prevap_resusp_optcc,& ! in
+                   precabx_old,    precabx_base_old,       & ! in
+                   scavabx_old,    precnumx_base_old,      & ! in
+                   precabx_new,    precabx_base_new,       & ! out
+                   scavabx_new,    resusp_x                ) ! out
+
+   elseif (evapx <= 0.0_r8) then
+      ! no evap so no resuspension
+      if ( mam_prevap_resusp_optcc <= 130) then
+         scavabx_new = scavabx_old
+      endif
+         resusp_x = 0.0_r8
```

Comment:
please fix the indent, someone like me may read `endif` as `else`:smirk:

Reply:
fixed